src/components/issue-event-list-item.component.js

```diff
@@ -0,0 +1,617 @@
+    const { actor, created_at: createdAt } = this.props.event;
+    const { action, onPressUser } = this.props;
+    const isRestored = action === 'restored';
+
+    return (
+      <Event
+        icon={
+          <EventIcon
+            name="git-branch"
+            backgroundColor={isRestored ? undefined : colors.greyBlue}
+            iconColor={isRestored ? undefined : colors.white}
+          />
+        }
+        text={
+          <Text style={{ padding: 3 }}>
+            <ActorLink actor={actor} onPress={onPressUser} /> {action} this
+            branch
+          </Text>
+        }
+        createdAt={createdAt}
+      />
+    );
+  }
+}
+
+class MarkedAsDuplicate extends Component {
+  props: {
+    event: Object,
+    unmarked: boolean,
+    onPressUser: Function,
+  };
+
+  render() {
+    // TODO: figure _which_ issue is the duplicate
+    const { actor, created_at: createdAt } = this.props.event;
+
+    return (
+      <Event
+        icon={
+          <EventIcon
+            name="bookmark"
+            backgroundColor={colors.greyBlue}
+            iconColor={colors.white}
+          />
+        }
+        text={
+          <Text style={{ padding: 3 }}>
+            <ActorLink actor={actor} onPress={this.props.onPressUser} /> marked
+            this as {this.props.unmarked ? 'not ' : ''}a duplicate
+          </Text>
+        }
+        createdAt={createdAt}
+      />
+    );
+  }
+}
+
+class ActorLink extends Component {
+  props: {
+    actor: Object,
+    onPress: Function,
+  };
+
+  render() {
+    const { actor, onPress } = this.props;
+
+    return (
+      <Text
+        style={styles.boldText}
+        onPress={() => {
+          onPress(actor);
+        }}
+      >
+        {actor.login}
+      </Text>
+    );
+  }
+}
+
+const marginLeftForIconName = name => {
+  switch (name) {
+    case 'git-branch':
+    case 'git-merge':
+    case 'primitive-dot':
+      return 8;
+    case 'bookmark':
+      return 6;
+    case 'person':
+    case 'lock':
+      return 4;
+    default:
+      return 2;
+  }
+};
+
+class EventIcon extends Component {
+  props: {
+    name: String,
+    iconColor: String,
+    backgroundColor: String,
+  };
+
+  render() {
+    const {
+      name,
+      iconColor = '#586069',
+      backgroundColor = '#E6EBF1',
+    } = this.props;
+
+    return (
+      <Icon
+        iconStyle={{
+          marginLeft: marginLeftForIconName(name),
+          marginTop: 1,
+          color: iconColor,
+        }}
+        containerStyle={[styles.iconContainer, { backgroundColor }]}
+        name={name}
+        type="octicon"
+        size={16}
+      />
+    );
+  }
+}
+
+class Date extends Component {
+  props: {
+    date: String,
+  };
+
+  render() {
+    const { date } = this.props;
+
+    return (
+      <View style={styles.dateContainer}>
+        <Text style={styles.date}>
+          {moment(date).fromNow()}
+        </Text>
+      </View>
+    );
+  }
+}
+
+class Bold extends Component {
```

Comment:
For `Date` and `Bold`, if you consider my suggestion to begin to use `EventInstance` or a component name similarly everywhere and if these components aren't being used more than once, we can probably remove their isolations. Otherwise this is fine :+1: